standalone.go

```diff
@@ -53,29 +63,87 @@ func (s *standalone) pick() int {
 	return rand.IntN(len(s.replicas))
 }
 
+func (s *standalone) handleRedirect(ctx context.Context, cmd Completed, result ValkeyResult) ValkeyResult {
+	if !s.enableRedirect {
+		return result
+	}
+
+	if ret, yes := IsValkeyErr(result.Error()); yes {
+		if addr, ok := ret.IsRedirect(); ok {
+			// Use singleflight to ensure only one redirect operation happens at a time
+			if err := s.redirectCall.Do(ctx, func() error {
+				return s.redirectToPrimary(addr)
```

Comment:
What if an ongoing singleflight mismatches with the `addr` here?

Reply:
I think Singleflight redirect should only proceed if the addr matches the latest seen redirect address

Reply:
I now think retries are all we need.